### PR TITLE
Fix jsx id bug when using interpolation in template string.

### LINKED ISSUE #107: Different components get the same jsx id if using interpolation.
For example:

``` javascript
// header.js
const Header = () => (
  <div className="root">
    <style jsx>{`
      .root {
        position: fixed;
        height: ${theme.height};
      }
    `}</style>
  </div>
);

// main.js
const Main = () => (
  <div className="root">
    <style jsx>{`
      .root {
        padding-top: ${theme.height};
      }
    `}</style>
  </div>
);
```

Both of them using interpolation in styles, and they get the same jsx id `[data-jsx="3533501187"]`.

### PR DESCRIPTION
**Pull request by lixiaoyan**
Fix #107

### CODE REVIEW

File: src/babel.js [hunk 1]
```diff
@@ -275,7 +275,7 @@ export default function ({types: t}) {
             ])
           }
 
-          state.jsxId = hash(state.styles.map(s => s[1]).join(''))
+          state.jsxId = hash(state.styles.map(s => s[1].source || s[1]).join(''))
```
Comment (giuseppeg) [inline src/babel.js]: maybe this could just be `hash(state.styles.reduce((h, s) => h+=s[0], ''))`

Comment (giuseppeg) [inline src/babel.js]: i.e. use `styleId` instead of `styleText` to generate the hash

Comment (giuseppeg) [inline src/babel.js]: that might blow up all the test and require us to refactor them, in that case maybe we can stick with your solution

**Comment by giuseppeg** [comment]
@lixiaoyan good catch and thanks for the PR, can you add a test for this use case?

**Comment by lixiaoyan** [comment]
@giuseppeg I think `test/fixtures/expressions.js` is enough, I have no idea to write a test for and only for this bug.

**Comment by giuseppeg** [comment]
> I have no idea to write a test for and only for this bug

@lixiaoyan something like your example from #108 should do it. They should render with different jsx id

**Comment by giuseppeg** [comment]
@lixiaoyan thank you!

**Comment by giuseppeg** [comment]
@nkzawa can you release 0.5.4 ?

**Comment by cgood92** [comment]
Thank you @lixiaoyan.... this has been messing me up for hours today, and I was just about to submit this same PR.  I pulled down the latest, saw that you had already done what I was just about to do, kicked myself for not checking, and then wanted to thank you 👍

**Comment by rauchg** [comment]
We've all been there @cgood92. Thank you for thinking about helping out `styled-jsx`!

**Pull request merged**